### DISCUSSION TITLE
Cannot get correct reason codes or reasoning strings

### DISCUSSION THREAD

**Discussion by PvanIeperen** [dotnet/MQTTnet#1532] [Q&A]
Hi,
When connecting or publishing to a broker, there will be a result. MqttClientConnectResult for connecting and MqttClientPublishResult for publishing. But the reason code always stay SUCCESS and the reasoning string always stay NULL.
Is there an example that shows what I have to do to get these values show the correct values?

regards Patrick

**Comment by chkr1011** [comment]
Please make sure to set the protocol version to 5.0.0.0 in the client options.
